src/Identity/Core/src/SignInManager.cs

```diff
@@ -432,6 +464,337 @@ public virtual async Task<SignInResult> CheckPasswordSignInAsync(TUser user, str
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Performs passkey assertion for the given <paramref name="credentialJson"/> and <paramref name="options"/>.
+    /// </summary>
+    /// <param name="credentialJson">The credentials obtained by JSON-serializing the result of the <c>navigator.credentials.get()</c> JavaScript function.</param>
+    /// <param name="options">The original passkey creation options provided to the browser.</param>
+    /// <returns>
+    /// A task object representing the asynchronous operation containing the <see cref="PasskeyAssertionResult{TUser}"/>.
+    /// </returns>
+    public virtual async Task<PasskeyAssertionResult<TUser>> PerformPasskeyAssertionAsync(string credentialJson, PasskeyRequestOptions options)
+    {
+        ThrowIfNoPasskeyHandler();
+        ArgumentException.ThrowIfNullOrEmpty(credentialJson);
+        ArgumentNullException.ThrowIfNull(options);
+
+        var user = options.UserId is { Length: > 0 } userId ? await UserManager.FindByIdAsync(userId) : null;
+        var context = new PasskeyAssertionContext<TUser>
+        {
+            User = user,
+            CredentialJson = credentialJson,
+            OriginalOptionsJson = options.AsJson(),
+            UserManager = UserManager,
+            HttpContext = Context,
+        };
+        var result = await _passkeyHandler.PerformAssertionAsync(context);
+        if (!result.Succeeded)
+        {
+            Logger.LogDebug(EventIds.PasskeyAssertionFailed, "Passkey assertion failed: {message}", result.Failure.Message);
+        }
+
+        return result;
+    }
+
+    [MemberNotNull(nameof(_passkeyHandler))]
+    private void ThrowIfNoPasskeyHandler()
+    {
+        if (_passkeyHandler is null)
+        {
+            throw new InvalidOperationException(
+                $"This operation requires an {nameof(IPasskeyHandler<>)} service to be registered.");
+        }
+    }
+
+    /// <summary>
+    /// Attempts to sign in the user with a passkey.
+    /// </summary>
+    /// <param name="credentialJson">The credentials obtained by JSON-serializing the result of the <c>navigator.credentials.get()</c> JavaScript function.</param>
+    /// <param name="options">The original passkey request options provided to the browser.</param>
+    /// <returns>
+    /// The task object representing the asynchronous operation containing the <see cref="SignInResult"/>
+    /// for the sign-in attempt.
+    /// </returns>
+    public virtual async Task<SignInResult> PasskeySignInAsync(string credentialJson, PasskeyRequestOptions options)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(credentialJson);
+
+        var assertionResult = await PerformPasskeyAssertionAsync(credentialJson, options);
+        if (!assertionResult.Succeeded)
+        {
+            return SignInResult.Failed;
+        }
+
+        var setPasskeyResult = await UserManager.SetPasskeyAsync(assertionResult.User, assertionResult.Passkey);
```

Comment:
This is the thing that updates the sign count on the database, isn't it?

Reply:
Yes, correct.